webapp/src/components/sidebar_right/sidebar_right.jsx

```diff
@@ -57,7 +69,25 @@ export default class SidebarRight extends React.PureComponent {
         switch (this.props.rhsState) {
         case RHSStates.PRS:
 
-            githubItems = this.props.yourPrs;
+            githubItems = this.props.yourPrs.map((pr) => {
+                let extra;
+                if (this.props.yourPrsExtraInfo) {
+                    extra = this.props.yourPrsExtraInfo.find((extraInfo) => {
+                        return (pr.repository_url === extraInfo.url) && (pr.number === extraInfo.number);
+                    });
+                }
+                if (!extra) {
+                    return pr;
+                }
+
+                return {
+                    ...pr,
+                    status: extra.status,
+                    reviewers: extra.reviewers,
+                    reviews: extra.reviews,
+                };
+            });
```

Comment:
Could we move this mapping logic to the reducer and have it ready when you map `yourPrsExtraInfo` for the state to the component? 

Also, these details would also be needed for the case of PR `RHSStates.REVIEWS` below. So this logic can be reused there for that case too.